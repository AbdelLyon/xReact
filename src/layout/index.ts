import "@/index.css";
export { Layout } from "@/layout/Layout.tsx";

export {
  type Navigation,
  type ProfileNavigation,
  type NavbarProps,
  type Language,
} from "@/types/layoutTypes";
