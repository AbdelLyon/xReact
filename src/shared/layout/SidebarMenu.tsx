import { cloneElement, isValidElement, ReactElement } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import { cn } from "@/utils";
import { Navigation, SidebarProps } from "@/types/layoutTypes";

interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
  className?: string;
  children: React.ReactNode;
}

export const SidebarMenu = ({
  navigation,
  pathname,
  className,
  isOpenDropdown,
  isNavbar,
}: Partial<SidebarProps>) => {
  const getLinkClassName = (item: Navigation) => {
    const isActive = pathname === item.pathname;

    return cn(
      "flex items-center justify-center p-3 cursor-pointer my-2 menu-transition transition-all duration-300 ease-in-out rounded-md line-transition w-full text-white lg:justify-start lg:gap-4",
      isNavbar ? "hover:bg-background" : "hover:bg-sidebarHoverLink",
      isActive ? "border-l border-primary" : "",
      isActive && (isNavbar ? "bg-background" : "bg-sidebarHoverLink"),
      { "justify-start gap-4 text-text": isOpenDropdown },
    );
  };

  return (
    <div className={className}>
      {navigation?.map(
        (item: Navigation) =>
          item.hasAccess && (
            <div key={item.name} className={getLinkClassName(item)}>
              {isValidElement(item.link)
                ? cloneElement(item.link as ReactElement<LinkProps>, {
                    className: cn(
                      "flex items-center w-full",
                      (item.link as ReactElement<LinkProps>).props.className,
                    ),
                    children: (
                      <div className="flex items-center gap-2">
                        {item.icon && (
                          <FontAwesomeIcon
                            className={cn("size-5", {
                              "text-primary": pathname === item.pathname,
                            })}
                            icon={item.icon}
                          />
                        )}
                        <span
                          className={cn("sr-only lg:not-sr-only", {
                            "not-sr-only": isOpenDropdown,
                          })}
                        >
                          {item.name}
                        </span>
                      </div>
                    ),
                  })
                : item.link}
            </div>
          ),
      )}
    </div>
  );
};
