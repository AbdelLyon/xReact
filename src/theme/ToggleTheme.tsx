import { IconMoonFilled, IconSunFilled } from "@tabler/icons-react";

import { cn } from "@/utils";
import { useTheme } from "@/hooks/useTheme";

export const ToggleTheme = ({ className }: { className?: string }) => {
  const { setTheme, theme } = useTheme();

  const handleClick = () => {
    const newTheme = theme === "light" ? "dark" : "light";

    setTheme(newTheme);
  };

  return (
    <button
      className={cn(
        "cursor-pointer transition-all hover:opacity-80 ",
        className,
      )}
      onClick={handleClick}
    >
      <IconSunFilled className="hidden dark:block" size={22} />
      <IconMoonFilled className="dark:hidden" size={22} />
    </button>
  );
};
