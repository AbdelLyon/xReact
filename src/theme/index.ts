import "@/index.css";
export { ToggleTheme } from "@/theme/ToggleTheme";
