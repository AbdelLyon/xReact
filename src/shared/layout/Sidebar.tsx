import { memo } from "react";

import { SidebarMenu } from "./SidebarMenu";

import { cn } from "@/utils";
import { SidebarProps } from "@/types/layoutTypes";

export const Sidebar = memo(
  ({
    navigation,
    pathname,
    className,
    isOpenDropdown,
    buttonSidebar,
  }: SidebarProps) => {
    return (
      <aside
        className={cn(
          "border-border bg-sidebar fixed inset-y-0 z-10 mt-16 flex min-h-screen",
          className,
        )}
      >
        <div className="sticky left-2 top-16 lg:fixed lg:w-[270px]">
          <nav className="mt-4 flex w-full flex-col items-center justify-center gap-4">
            {buttonSidebar?.()}
            <SidebarMenu
              className="w-full"
              isOpenDropdown={isOpenDropdown}
              navigation={navigation}
              pathname={pathname}
            />
          </nav>
        </div>
      </aside>
    );
  },
);

Sidebar.displayName = "Sidebar";
