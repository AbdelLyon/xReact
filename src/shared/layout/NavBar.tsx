import { forwardRef, useEffect, useRef, memo } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { AlignLeft, User } from "lucide-react";

import { SelectLanguage } from "../select/SelectLanguage";

import { SidebarMenu } from "./SidebarMenu";

import { ToggleTheme } from "@/theme/ToggleTheme";
import { cn } from "@/utils";
import { NavbarProps, ProfileNavigation } from "@/types/layoutTypes";
import { Popover } from "@/popover/Popover";
import { PopoverClose } from "@/ui/popover";

const MenuBarNavigation = memo(
  ({ navigation = [] }: { navigation: ProfileNavigation[] }) => (
    <nav className="border border-border bg-white p-2 shadow-none dark:bg-card">
      {navigation.map((item) => {
        const navItem = typeof item === "string" ? { name: item } : item;

        return (
          <PopoverClose
            key={item.name}
            className="flex w-full items-center gap-2 rounded-md p-2 transition-colors hover:bg-gray-100 dark:text-white dark:hover:bg-background"
            onClick={navItem.onClick}
          >
            {navItem.icon && (
              <FontAwesomeIcon className="size-4" icon={navItem.icon} />
            )}

            <span>{navItem.name}</span>
          </PopoverClose>
        );
      })}
    </nav>
  ),
);

MenuBarNavigation.displayName = "MenuBarNavigation";

const UserPopover = memo(
  ({
    currentUser,
    profileNavigation,
  }: {
    currentUser: string;
    profileNavigation: ProfileNavigation[];
    onOpenChange?: (open: boolean) => void;
    open?: boolean;
  }) => {
    return (
      <Popover
        classNames={{ popover: "w-[200px]", trigger: "border-none" }}
        trigger={
          <>
            <div className="relative hidden cursor-pointer items-center gap-2 rounded-md px-3 py-2 font-medium dark:text-white lg:flex">
              <User className="size-5" />
              <p className="truncate">{currentUser}</p>
            </div>
            <div className="flex size-10 cursor-pointer items-center justify-center rounded-md dark:bg-background dark:text-white lg:hidden">
              <User className="size-5" />
            </div>
          </>
        }
      >
        <MenuBarNavigation navigation={profileNavigation} />
      </Popover>
    );
  },
);

UserPopover.displayName = "UserPopover";

const MobileMenu = forwardRef<
  HTMLDivElement,
  {
    isOpenDropdown: boolean;
    mobileNavigation: any;
    pathname: string;
    buttonAction?: () => React.ReactNode;
  }
>(({ isOpenDropdown, mobileNavigation, pathname, buttonAction }, ref) => (
  <div
    ref={ref}
    className={cn(
      "flex flex-col gap-2 overflow-hidden transition-all duration-300 ease-in-out",
      isOpenDropdown
        ? "opacity-100 max-h-[var(--navbar-content-height)]"
        : "opacity-0 max-h-0",
    )}
  >
    <SidebarMenu
      isNavbar
      className="w-full"
      isOpenDropdown={isOpenDropdown}
      navigation={mobileNavigation}
      pathname={pathname}
    />
    {buttonAction && <div className="mb-4">{buttonAction()}</div>}
  </div>
));

MobileMenu.displayName = "MobileMenu";

export const Navbar = memo(
  ({
    applicationName,
    currentUser,
    profileNavigation,
    className,
    setOpenDropdown,
    isOpenDropdown,
    mobileNavigation,
    pathname,
    logo,
    buttonAction,
    languages,
    languageSelected,
    onLanguageChange,
    notification,
  }: NavbarProps) => {
    const contentRef = useRef<HTMLDivElement>(null);

    useEffect(() => {
      if (contentRef.current) {
        const height = isOpenDropdown ? contentRef.current.scrollHeight : 0;

        document.documentElement.style.setProperty(
          "--navbar-content-height",
          `${height}px`,
        );
      }
    }, [isOpenDropdown]);

    const headerClasses = cn(
      "fixed top-0 z-40 flex w-full flex-col border-b border-border bg-white dark:bg-sidebar shadow-lg dark:shadow-none px-4 transition-all duration-300 ease-in-out",
      isOpenDropdown ? "h-[calc(var(--navbar-content-height)+4rem)]" : "h-16",
      className,
    );

    const renderLogo = (className: string) =>
      logo && <div className={className}>{logo()}</div>;

    const renderMobileMenuButton = () => (
      <button
        className="rounded-md border border-border bg-white px-2.5 py-2 dark:bg-background sm:hidden"
        onClick={() => setOpenDropdown?.(!isOpenDropdown)}
      >
        <AlignLeft className="text-text size-5" />
      </button>
    );

    const renderNavItems = () => (
      <nav className="flex items-center justify-end gap-4 rounded-md border border-border bg-white dark:bg-background">
        {languages && (
          <SelectLanguage
            languages={languages}
            value={languageSelected?.value ?? ""}
            onLanguageChange={onLanguageChange}
          />
        )}
        <ToggleTheme className="dark:text-white" />
        <div className="h-10 w-[2px] bg-border" />
        {notification && (
          <FontAwesomeIcon
            className="size-4 cursor-pointer dark:text-white"
            icon={notification.icon}
            onClick={notification.onClick}
          />
        )}
        <UserPopover
          currentUser={currentUser}
          profileNavigation={profileNavigation}
        />
      </nav>
    );

    return (
      <header className={headerClasses}>
        <div className="flex h-16 w-full items-center justify-between">
          {renderLogo(
            "border-border w-[275px] justify-center border-r hidden lg:block",
          )}
          {renderMobileMenuButton()}
          <div className="flex w-full flex-1 items-center justify-end text-black lg:pl-8">
            {applicationName ? (
              <h1 className="hidden w-full cursor-default select-none text-2xl font-semibold dark:text-white sm:block">
                {applicationName}
              </h1>
            ) : (
              renderLogo(
                "border-border hidden sm:flex w-full justify-start border-r lg:hidden",
              )
            )}
            {renderNavItems()}
          </div>
        </div>
        <MobileMenu
          ref={contentRef}
          buttonAction={buttonAction}
          isOpenDropdown={isOpenDropdown || false}
          mobileNavigation={mobileNavigation}
          pathname={pathname}
        />
      </header>
    );
  },
);
Navbar.displayName = "Navbar";
