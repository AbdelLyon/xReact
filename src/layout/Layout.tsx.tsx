import { useState, useCallback, useEffect } from "react";

import { Navbar } from "@/shared/layout/NavBar";
import { Sidebar } from "@/shared/layout/Sidebar";
import { DailyAppsLayoutProps } from "@/types/layoutTypes";
import { cn } from "@/utils";

export const Layout = ({
  header,
  children,
  className,
  sidebar,
}: DailyAppsLayoutProps) => {
  const [isOpen, setIsOpen] = useState(false);

  const setOpenDropdown = useCallback((isOpen: boolean) => {
    setIsOpen(isOpen);
  }, []);

  useEffect(() => {
    const mediaQuery = window.matchMedia("(min-width: 768px)");

    const handleMediaQueryChange = () => setIsOpen(false);

    if (mediaQuery.matches) setIsOpen(false);

    mediaQuery.addEventListener("change", handleMediaQueryChange);

    return () => {
      mediaQuery.removeEventListener("change", handleMediaQueryChange);
    };
  }, []);

  const sidebarProps = {
    buttonSidebar: sidebar.buttonSidebar,
    className: cn(
      "w-16 px-1.5 flex-col lg:px-3 border-r transition-all duration-300 ease-in-out",
      "lg:w-[290px] lg:min-w-[290px] lg:mt-0 md:sticky",
      "hidden sm:flex",
      className,
    ),
    isOpenDropdown: isOpen,
    navigation: sidebar.navigation,
    pathname: sidebar.pathname,
  };

  return (
    <div className=" min-h-screen w-full overflow-x-auto">
      <Navbar
        ToggleTheme={header.ToggleTheme}
        applicationName={header.applicationName}
        buttonAction={header.buttonAction}
        className={header.className}
        currentUser={header.currentUser}
        isOpenDropdown={isOpen}
        languageSelected={header.languageSelected}
        languages={header.languages}
        logo={header.logo}
        mobileNavigation={header.mobileNavigation}
        notification={header.notification}
        pathname={header.pathname}
        profileNavigation={header.profileNavigation}
        setOpenDropdown={setOpenDropdown}
        onLanguageChange={header.onLanguageChange}
      />
      <div className="relative z-20 flex w-full">
        {!isOpen && <Sidebar {...sidebarProps} />}

        <main
          className={cn(
            "mx-auto w-full mt-20 mb-4 sm:ml-16 md:ml-0",
            className,
          )}
        >
          {children}
        </main>

        {isOpen && (
          <button
            className="fixed inset-0 bg-black/20 backdrop-blur-sm"
            onClick={() => setIsOpen(false)}
          />
        )}
      </div>
    </div>
  );
};
