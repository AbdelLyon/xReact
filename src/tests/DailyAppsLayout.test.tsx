import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { faHome } from "@fortawesome/free-solid-svg-icons";
import { AlignLeft } from "lucide-react";

import { Layout } from "@/layout";
import { ToggleTheme } from "@/theme";

// Mock props for Layout
const mockProps = {
  header: {
    Logo: <div>App Logo</div>,
    profileNavigation: [
      {
        name: "Profile",
        icon: faHome,
        onClick: () => {},
      },
      {
        name: "Settings",
        icon: faHome,
        onClick: () => {},
      },
    ],
    applicationName: "My App",
    className: "header-class",
    navigation: [
      {
        name: "Home",
        link: <a href="/">/home</a>,
        pathname: "/",
        icon: faHome,
        hasAccess: true,
      },
    ],
    pathname: "/",
    toggleTheme: () => {},
    user: "John Doe",
    buttonNavbar: () => <button />,
  },
  sidebar: {
    buttonSidebar: "Menu",
    btnIcon: <AlignLeft />,
    className: "sidebar-class",
    navigation: [
      {
        name: "Dashboard",
        link: <a href="/dashboard">/dashboard</a>,
        pathname: "/dashboard",
        icon: faHome,
        hasAccess: true,
      },
    ],
    pathname: "/",
  },
  children: <div>Content goes here</div>,
  className: "layout-class",
};

// Snapshot test for Layout
describe("Layout", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <Layout
        className={mockProps.className}
        header={{
          ...mockProps.header,
          toggleTheme: true,
          logo: () => mockProps.header.Logo,
          currentUser: "",
          mobileNavigation: [],
          ToggleTheme: ToggleTheme,
        }}
        sidebar={{
          ...mockProps.sidebar,
          buttonSidebar: () => mockProps.sidebar.buttonSidebar,
        }}
      >
        {mockProps.children}
      </Layout>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
