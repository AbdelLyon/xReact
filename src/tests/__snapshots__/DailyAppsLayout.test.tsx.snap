// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class=" min-h-screen w-full overflow-x-auto"
  >
    <header
      class="fixed top-0 z-40 flex w-full flex-col border-b border-border bg-white dark:bg-sidebar shadow-lg dark:shadow-none px-4 transition-all duration-300 ease-in-out h-16 header-class"
    >
      <div
        class="flex h-16 w-full items-center justify-between"
      >
        <div
          class="border-border w-[275px] justify-center border-r hidden lg:block"
        >
          <div>
            App Logo
          </div>
        </div>
        <button
          class="rounded-md border border-border bg-white px-2.5 py-2 dark:bg-background sm:hidden"
        >
          <svg
            class="lucide lucide-align-left size-5 text-text"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="21"
              x2="3"
              y1="6"
              y2="6"
            />
            <line
              x1="15"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="17"
              x2="3"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
        <div
          class="flex w-full flex-1 items-center justify-end text-black lg:pl-8"
        >
          <h1
            class="hidden w-full cursor-default select-none text-2xl font-semibold dark:text-white sm:block"
          >
            My App
          </h1>
          <nav
            class="flex items-center justify-end gap-4 rounded-md border border-border bg-white dark:bg-background"
          >
            <button
              class="cursor-pointer transition-all hover:opacity-80 dark:text-white"
            >
              <svg
                class="lucide lucide-sun hidden dark:block"
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <path
                  d="M12 2v2"
                />
                <path
                  d="M12 20v2"
                />
                <path
                  d="m4.93 4.93 1.41 1.41"
                />
                <path
                  d="m17.66 17.66 1.41 1.41"
                />
                <path
                  d="M2 12h2"
                />
                <path
                  d="M20 12h2"
                />
                <path
                  d="m6.34 17.66-1.41 1.41"
                />
                <path
                  d="m19.07 4.93-1.41 1.41"
                />
              </svg>
              <svg
                class="lucide lucide-moon dark:hidden"
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
                />
              </svg>
            </button>
            <div
              class="h-10 w-[2px] bg-border"
            />
            <div
              aria-controls="radix-:r0:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="border border-border rounded-md border-none"
              data-state="closed"
              type="button"
            >
              <div
                class="relative hidden cursor-pointer items-center gap-2 rounded-md px-3 py-2 font-medium dark:text-white lg:flex"
              >
                <svg
                  class="lucide lucide-user size-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
                  />
                  <circle
                    cx="12"
                    cy="7"
                    r="4"
                  />
                </svg>
                <p
                  class="truncate"
                />
              </div>
              <div
                class="flex size-10 cursor-pointer items-center justify-center rounded-md dark:bg-background dark:text-white lg:hidden"
              >
                <svg
                  class="lucide lucide-user size-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
                  />
                  <circle
                    cx="12"
                    cy="7"
                    r="4"
                  />
                </svg>
              </div>
            </div>
          </nav>
        </div>
      </div>
      <div
        class="flex flex-col gap-2 overflow-hidden transition-all duration-300 ease-in-out opacity-0 max-h-0"
      >
        <div
          class="w-full"
        />
      </div>
    </header>
    <div
      class="relative z-20 flex w-full"
    >
      <aside
        class="border-border bg-sidebar fixed inset-y-0 z-10 mt-16 min-h-screen w-16 px-1.5 flex-col lg:px-3 border-r transition-all duration-300 ease-in-out lg:w-[290px] lg:min-w-[290px] lg:mt-0 md:sticky hidden sm:flex layout-class"
      >
        <div
          class="sticky left-2 top-16 lg:fixed lg:w-[270px]"
        >
          <nav
            class="mt-4 flex w-full flex-col items-center justify-center gap-4"
          >
            Menu
            <div
              class="w-full"
            >
              <div
                class="flex items-center justify-center p-3 cursor-pointer my-2 menu-transition transition-all duration-300 ease-in-out rounded-md line-transition w-full text-white lg:justify-start lg:gap-4 hover:bg-sidebarHoverLink"
              >
                <a
                  class="flex items-center w-full"
                  href="/dashboard"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-house size-5"
                      data-icon="house"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      class="sr-only lg:not-sr-only"
                    >
                      Dashboard
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </nav>
        </div>
      </aside>
      <main
        class="mx-auto w-full mt-20 mb-4 sm:ml-16 md:ml-0 layout-class"
      >
        <div>
          Content goes here
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
